boefjes/boefjes/sql/session.py

```diff
@@ -34,15 +34,13 @@ def __exit__(self, exc_type: type[Exception], exc_value: str, exc_traceback: str
 
             return
 
-        error = None
-
         try:
             logger.debug("Committing session")
             self.session.commit()
-        except DatabaseError as e:
-            error = e
+        except exc.IntegrityError as e:
```

Comment:
Is there no more specific error for the uniqueConstraint being thrown? I see psycog2 has one specifically for that.

Reply:
I fixed this partly and double-checked the documentation, but the best we could do getting the field name out of the message is parsing the exception string I'm afraid